### PR TITLE
community: Add Dria retriever

### PR DESCRIPTION
**Pull request by anilaltuner**
[Dria](https://dria.co/) is a hub of public RAG models for developers to both contribute and utilize a shared embedding lake. This PR adds a retriever that can retrieve documents from Dria.

### CODE REVIEW

**Comment by anilaltuner** [comment]
Hey @baskaryan, it's ready for review.

**Comment by anilaltuner** [comment]
@hwchase17

File: libs/community/pyproject.toml [hunk 1]
```diff
@@ -92,6 +92,7 @@ hdbcli = {version = "^2.19.21", optional = true}
 oci = {version = "^2.119.1", optional = true}
 rdflib = {version = "7.0.0", optional = true}
 nvidia-riva-client = {version = "^2.14.0", optional = true}
+dria = {version = "^0.1.7", optional = true}
```
Comment (baskaryan) [inline libs/community/pyproject.toml:95]: shouldn't be added unless it's needed for unit tests

File: libs/community/langchain_community/utilities/dria_index.py [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+    Attributes:
+        api_key: Your API key for accessing Dria.
+        contract_id: The contract ID of the knowledge base to interact with.
+        top_n: Number of top results to fetch for a search.
+    """
+
+    def __init__(
+        self, api_key: str, contract_id: Optional[str] = None, top_n: int = 10
+    ):
+        from dria import Dria, Models
```
Comment (baskaryan) [inline libs/community/langchain_community/utilities/dria_index.py:23]: can we try/except this import with a nice error message telling you how to install it

File: libs/community/langchain_community/utilities/dria_index.py [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+        self.dria_client = Dria(api_key=self.api_key)
+        if self.contract_id:
+            self.dria_client.set_contract(self.contract_id)
+
+    def create_knowledge_base(
+        self,
+        name: str,
+        description: str,
+        category: str = "Unspecified",
+        embedding: str = "jina-embeddings-v2-base-en",
```
Comment (baskaryan) [inline libs/community/langchain_community/utilities/dria_index.py:38]: will users usually want to customize this? in that case may be clearer to not give it a default

File: libs/community/langchain_community/retrievers/dria_index.py [hunk 4]
```diff
@@ -0,0 +1,87 @@
...
+    def __init__(self, api_key: str, contract_id: Optional[str] = None):
+        """
+        Initialize the DriaRetriever with a DriaAPIWrapper instance.
+
+        Args:
+            api_key: The API key for Dria.
+            contract_id: The contract ID of the knowledge base to interact with.
+        """
+        super().__init__()
+        self.api_wrapper = DriaAPIWrapper(api_key=api_key, contract_id=contract_id)
```
Comment (baskaryan) [inline libs/community/langchain_community/retrievers/dria_index.py:26]: should flip this order
```python
def __init__(self, api_key: str, *, contract_id: Optional[str]=None, **kwargs: Any):
    api_wrapper = DriaAPIWrapper(...)
    super().__init__(api_wrapper=api_wrapper, **kwargs)
```

**Review by baskaryan — COMMENTED** [review]
very cool! few comments

**Comment by anilaltuner** [comment]
Thanks @baskaryan for your review! I will agree on all of them and release extra commit with fixes for import issues about Dria. I think its all right now. Waiting for workflow appoval

**Comment by anilaltuner** [comment]
Hey @baskaryan? Can you review for workflow?

File: libs/langchain/langchain/retrievers/__init__.py
Comment (baskaryan) [inline libs/langchain/langchain/retrievers/__init__.py:1]: can we remove all the changes to `libs/langchain`? these imports exist only for backwards compatibility, which isn't an issue for new integrations

Comment (anilaltuner) [reply libs/langchain/langchain/retrievers/__init__.py:1]: Got it, all removed

**Comment by anilaltuner** [comment]
Thanks for reviews @baskaryan. Should ready for workflow approval

**Comment by anilaltuner** [comment]
@baskaryan Hello, Is there an any issue on merging this pr?

**Comment by anilaltuner** [comment]
Conflicts with the new release resolved

**Comment by anilaltuner** [comment]
Hey @baskaryan! This integration unlocks many of our use cases and open source projects. Can we manage reviews and merges? Thanks for your time.

**Review by baskaryan — APPROVED** [review]

**Comment by anilaltuner** [comment]
Hey @baskaryan! Vercel deployment on workflow and cannot see the logs. Can you give me detail on that?

**Pull request merged**